napalm/base/helpers.py

```diff
@@ -232,13 +257,23 @@ def find_txt(xml_tree, path, default=""):
     value = ""
     try:
         xpath_applied = xml_tree.xpath(path)  # will consider the first match only
-        if len(xpath_applied) and xpath_applied[0] is not None:
+        xpath_length = len(xpath_applied)  # get a count of items in XML tree
+        if xpath_length and xpath_applied[0] is not None:
             xpath_result = xpath_applied[0]
             if isinstance(xpath_result, type(xml_tree)):
                 value = xpath_result.text.strip()
             else:
                 value = xpath_result
-    except Exception:  # in case of any exception, returns default
+        else:
+            if xpath_applied == "":
+                xpath_applied = ""
```

Comment:
Did you intend to have something different here?

Reply:
Nice catch. Took me a minute to get what you were driving at.

I originally had a if `xpath_applied == None or xpath_applied== ""` but I think python black didn't like that when the code was being checked. I wanted to do conditionals to express that the variable could be an empty string  or just not be set at all. Next push will feature just the debug message when xpath is applied.